### PR TITLE
Create GitHub Action to auto-publish to PyPI

### PR DESCRIPTION
**Pull request by libialany**
- Configure repository secrets for user and password of PyPI authentication: ``user: ${{ secrets.pypi_user }}`` ``password: ${{ secrets.pypi_password }}``.
- Implement package uploads exclusively for tagged commits using the [gh-action-pypi-publish](https://github.com/pypa/gh-action-pypi-publish/blob/unstable/v1/README.md).
- Integrate the [isolation feature](https://packaging.python.org/en/latest/guides/modernize-setup-py-project/) by adding the pyproject.toml file.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        # linux
+        - os: ubuntu-latest
```
Comment (whimboo) [inline .github/workflows/release.yml:16]: We actually do not have any platform specific code included. As such we do not need separate build artifacts for the release. It will be enough to provide the tar'ed source package and the wheel. That means it will be enough to run this job for Linux and for the latest supported Python version.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Install dependencies
+        run: python -m pip install -U build
+
+      - name: Build package
+        run: python -m build    
+
+      - name: Publish package to PyPI
```
Comment (whimboo) [inline .github/workflows/release.yml:82]: I would suggest to split the jobs into `build` and `pypi-publish` similar to the example on https://github.com/marketplace/actions/pypi-publish. Maybe you can update the related code as well to better match

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        if: github.repository == 'mozilla/mozdownload' && github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          user: ${{ secrets.pypi_user }}
+          repository-url: https://test.pypi.org/legacy/
+          password: ${{ secrets.pypi_password }}
```
Comment (whimboo) [inline .github/workflows/release.yml:88]: Validation has to happen via a token and not by using a known user. That will make it more secure and easier to update or revoke permissions.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+          python-version: ${{ matrix.python }}
+
+      - name: Install dependencies
+        run: python -m pip install -U build
+
+      - name: Build package
+        run: python -m build    
+
+      - name: Publish package to PyPI
+        if: github.repository == 'mozilla/mozdownload' && github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
```
Comment (whimboo) [inline .github/workflows/release.yml:83]: Could we change the check for `mozilla/mozdownload` to the existence of the secret token identifier in the environment?

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+[build-system]
+requires = ['setuptools>=40']
```
Comment (whimboo) [inline pyproject.toml:2]: For what specifically is this file needed?

File: .github/workflows/release.yml (hunk 1 above)
Comment (libialany) [reply .github/workflows/release.yml]: Thank you, I had that question.

File: .github/workflows/release.yml (hunk 2 above)
Comment (libialany) [reply .github/workflows/release.yml]: Thank you, I'll be creating two jobs.

File: .github/workflows/release.yml (hunk 4 above)
Comment (libialany) [reply .github/workflows/release.yml]: Thank you for the advice.

File: pyproject.toml (hunk 5 above)
Comment (libialany) [reply pyproject.toml:2]: It's used to configure setuptools with [pyproject.toml](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html), but it's not necessary.

**Comment by whimboo** [comment]
FYI a new PR was created for the requested changes which is #686.
